File: GXpert/GXpert.Web/Modules/Schools/SchoolTeacher/SchoolTeacherColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Schools.Columns;

[ColumnsScript("Schools.SchoolTeacher")]
[BasedOnRow(typeof(SchoolTeacherRow), CheckNames = true)]
public class SchoolTeacherColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string TeacherPrn { get; set; }
    public string SchoolName { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
    [EditLink]
    public string Description { get; set; }
    public string AcademicYearName { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolTeacherColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SchoolTeacherRow } from "./SchoolTeacherRow";

export interface SchoolTeacherColumns {
    Id: Column<SchoolTeacherRow>;
    TeacherPrn: Column<SchoolTeacherRow>;
    SchoolName: Column<SchoolTeacherRow>;
    ClassTitle: Column<SchoolTeacherRow>;
    SubjectTitle: Column<SchoolTeacherRow>;
    Description: Column<SchoolTeacherRow>;
    AcademicYearName: Column<SchoolTeacherRow>;
    InsertDate: Column<SchoolTeacherRow>;
    InsertUserId: Column<SchoolTeacherRow>;
    UpdateDate: Column<SchoolTeacherRow>;
    UpdateUserId: Column<SchoolTeacherRow>;
    IsActive: Column<SchoolTeacherRow>;
}

export class SchoolTeacherColumns extends ColumnsBase<SchoolTeacherRow> {
    static readonly columnsKey = 'Schools.SchoolTeacher';
    static readonly Fields = fieldsProxy<SchoolTeacherColumns>();
}